packages/block-library/src/embed/edit.js

```diff
@@ -52,12 +59,36 @@ export function getEmbedEditComponent( title, icon, responsive = true ) {
 			}
 		}
 
+		handlePaste() {
+			const { url } = this.props.attributes;
+			const { fetching, cannotEmbed } = this.props;
+
+			// This URL can't be embedded, so replace with the fallback block.
+			if ( url && ! fetching && cannotEmbed ) {
+				this.props.onReplace(
+					createPasteFallbackBlock( this.props.attributes.url )
+				);
+				return;
+			}
+
+			// This URL can be embedded, so pass it over to the default embed block to continue.
+			if ( url && ! fetching && ! cannotEmbed ) {
+				this.props.onReplace(
+					createBlock( DEFAULT_EMBED_BLOCK, { url } )
```

Comment:
If we would include this paste logic in the default block, we could probably skip this part.